src/module/Boundary.cpp

```diff
@@ -40,6 +40,58 @@ std::string PeriodicBox::getDescription() const {
 	return s.str();
 }
 
+
+ReflectiveShell::ReflectiveShell(Vector3d cen, double r) :
+		center(cen), radius(r) {
+		}
+
+double ReflectiveShell::distance(const Vector3d &point) const {
+	Vector3d dR = point - center;
+	return dR.getR() - radius;
+}
+
+Vector3d ReflectiveShell::normal(const Vector3d& point) const {
+	Vector3d d = point - center;
+	return d.getUnitVector();
+}
+
+void ReflectiveShell::process(Candidate *c) const {
+	double currentDistance = distance(c->current.getPosition());
+	double previousDistance = distance(c->previous.getPosition());
+	// check if cosmic ray crossed boundary in last step
+	if (currentDistance * previousDistance < 0){
+		Vector3d currentDirection = c->current.getDirection();
+		Vector3d previousPosition = c->previous.getPosition();
+		// get point where trajectory intersects the shell boundary
+		double p_half = previousPosition.dot(currentDirection) / currentDirection.getR2();
+		double q = (previousPosition.getR2() - radius*radius) / currentDirection.getR2();
```

Comment:
Code Style: spaces around operators. Use `a * b` instead of `a*b`. 
This also happens below. I mark it only once here.